lib/Github/Api/Repo.php

```diff
@@ -535,6 +535,26 @@ public function branches($username, $repository, $branch = null, array $paramete
         return $this->get($url, $parameters);
     }
 
+
+    /**
+     * Sync a fork branch with the upstream repository
+     *
+     * @link https://docs.github.com/en/rest/branches/branches#sync-a-fork-branch-with-the-upstream-repository
+     *
+     * @param $username
+     * @param $repository
+     * @param $branchName
+     *
+     * @return array|string
+     */
+    public function syncUpstream($username, $repository, $branchName = null)
```

Comment:
This is tough to name, the API route is `merge-upstream` so folks would look for that, but the description is `sync`.